src/android/Peripheral.java

```diff
@@ -225,10 +233,6 @@ public void onConnectionStateChange(BluetoothGatt gatt, int status, int newState
             gatt.discoverServices();
 
         } else {
-
-            if (connectCallback != null) {
```

Comment:
The error callback needs to happen here so it's only called when the peripheral disconnects

This function no longer calls, disconnect() so the read and write failure callbacks probably need to do here. I need to think about that a bit.